components/layout_2020/flexbox/layout.rs

```diff
@@ -158,22 +158,26 @@ impl FlexContainer {
         // Absolutely-positioned children of the flex container may be interleaved
         // with flex items. We need to preserve their relative order for correct painting order,
         // which is the order of `Fragment`s in this function’s return value.
-        let original_order_with_absolutely_positioned = self
+        //
+        // Example:
+        // absolutely_positioned_items_with_original_order = [Some(item), Some(item), None, Some(item), None]
+        // flex_items                                      =                         [item,             item]
+        let absolutely_positioned_items_with_original_order = self
             .children
             .iter()
             .map(|arcrefcell| {
                 let borrowed = arcrefcell.borrow_mut();
                 match &*borrowed {
                     FlexLevelBox::OutOfFlowAbsolutelyPositionedBox(absolutely_positioned) => {
-                        Ok(absolutely_positioned.clone())
+                        Some(absolutely_positioned.clone())
                     },
                     FlexLevelBox::FlexItem(_) => {
                         let item = AtomicRefMut::map(borrowed, |child| match child {
                             FlexLevelBox::FlexItem(item) => item,
                             _ => unreachable!(),
                         });
                         flex_items.push(item);
-                        Err(())
+                        None
```

Comment:
If we are cleaning up this code, I think it might better for this code to stop using `Result` and `Option` to track these variables and use an appropriately named enum in order to be as clear as possible. Perhaps something like this:

```rust
enum FlexContent {
    AbsolutelyPositionedBox(...)
    FlexItemPlaceholder
}
```

Reply:
I love that idea

Reply:
I played around with other ideas of holding all flex items in that enum you suggested, instead of just the absolutely positioned ones, just to see if I could take the simplification a step further. However, I found that to be more confusing. Having a vector of enums with data and placeholders, then another vector with items to fill the placeholders is the simplest way I can think of (this is the current implementation). I think having a dedicated enum for this, like we have it in this PR, will make the code a bit friendlier to read.

Ready for another review!